### ISSUE TITLE
"Your mono runtime and class libraries are out of sync message" on building unit test projects with XI master

### ISSUE THREAD

**Issue by GouriKumari** [xamarin/xamarin-macios#4174]
### Steps to Reproduce

1. Update System with Xamarin.iOS project, XI version: 11.11.0.331 (master: 238eb94c) 
2. Build and Run any unit test on device.

### Expected Behavior
Build Log should not contain any out of sync messages.

### Actual Behavior
"Your mono runtime and class libraries are out of sync" message is  present in build logs.

### Environment

```
Operating System: Mac OS X 10.13.4
Bot Name: XQA-XI-Sierra-6
Mono version: 5.14.0.106 (820bff26b3b)  (2018-05-25 17:29:19 -04:00)
XA version: 8.3.99.19 (HEAD/09df99bdc) 
XI version: 11.11.0.331 (master: 238eb94c) 
XM version: 4.5.0.331 (master: 238eb94c)  (2018-05-25 07:47:00 -04:00)
XVS version: 
XS version: 7.6.0.971 (e936c76)  (2018-05-28 12:49:41 -04:00)
```

### Build Logs

http://xqa.blob.core.windows.net/gist/report-c8b3a0e9db67473ebe1ff9df22e7c44d.txt, with mono 5.14
http://xqa.blob.core.windows.net/gist/report-641ce6bd5c0545618ad5b7f981fcbbdf.txt with mono 5.12

### Example Project (If Possible)

Any unit test project in xamarin-macios/tests/ folder. 

## Additional Info:
This is not a new issue in master. It was happening for quite sometime. I confirmed that QA tests are checking out the correct external/mono and build logs are not giving false results, also checked manually. This issue is occurring in bots with either mono 5.14 or 5.12 installed.

**Comment by spouliot** [comment]
```
  Mono Ahead of Time compiler - compiling assembly /Users/xamarinqa/agent/_work/r19/a/xamarin-macios/tests/monotouch-test/obj/iPhone/Debug-unified/mtouch-cache/64/3-Build/mscorlib.dll
  AOTID B67D7582-90F6-4BAF-C781-92CED65C2A7D
  cant resolve internal call to "System.Runtime.Remoting.RemotingServices::InternalExecute" (tested without signature also)
  
  Your mono runtime and class libraries are out of sync.
  The out of sync library is: /Users/xamarinqa/agent/_work/r19/a/xamarin-macios/tests/monotouch-test/obj/iPhone/Debug-unified/mtouch-cache/64/3-Build/mscorlib.dll
  
  When you update one from git you need to update, compile and install
  the other too.
  Do not report this as a bug unless you're sure you have updated correctly:
  you probably have a broken mono install.
  If you see other errors or faults after this message they are probably related
  and you need to fix your mono install first.
  ```

@vargaz with the SDK changes you're building both the runtime and BCL right ?

Can it be an icall that is not is in mscorlib.dll but removed from the (iOS optimized) AOT compiler build ?

**Comment by vargaz** [comment]
InternalExecute is not present because the runtime is compiled using DISABLE_REMOTING, mscorlib is supposed to be compiled using the same define.

**Comment by vargaz** [comment]
Not sure what caused this. The compiled mscorlib still has the private InternalGetProxyType () icall, even through nothing references it. Maybe a linker problem ?

**Comment by marek-safar** [comment]
This is indeed linker setup issue. I'll look into fixing this

**Comment by vargaz** [comment]
I can make this into an assert if it helps.

**Comment by borrrden** [comment]
What release should this be in?  I see this in 11.14.0.14

**Comment by rolfbjarne** [comment]
@borrrden this fix has not been released yet (most likely it will be in our 12.2 release, scheduled for some time late this fall).

Additionally there's another source of this warning (for a different function) that's not fixed yet (#4608), so even with the 12.2 release you'll probably see the warning (just fewer of them).
